include/vara/Feature/FeatureModelWriter.h

```diff
@@ -0,0 +1,61 @@
+#ifndef VARA_FEATURE_FEATUREMODELWRITER_H
+#define VARA_FEATURE_FEATUREMODELWRITER_H
+
+#include "vara/Feature/FeatureModel.h"
+
+#include "libxml/tree.h"
+#include "libxml/xmlwriter.h"
+
+#include <memory>
+
+namespace vara::feature {
+
+//===----------------------------------------------------------------------===//
+//                               FeatureModelParser Class
+//===----------------------------------------------------------------------===//
+
+/// \brief Base class for parsers with different input formats.
+class FeatureModelWriter {
+protected:
+  explicit FeatureModelWriter() = default;
+
+public:
+  virtual ~FeatureModelWriter() = default;
+
+  /// Build \a FeatureModel after parsing. May return null if parsing or
+  /// building failed.
+  ///
+  /// \returns an instance of \a FeatureModel or \a nullptr
+  virtual int writeFeatureModel(std::string Path) = 0;
+
+};
+
+//===----------------------------------------------------------------------===//
+//                               FeatureModelXmlParser Class
+//===----------------------------------------------------------------------===//
+
+/// \brief Parsers for feature models in XML.
+class FeatureModelXmlWriter : public FeatureModelWriter {
+public:
+  explicit FeatureModelXmlWriter(const FeatureModel &Fm) : Fm{Fm}{}
+
+  int writeFeatureModel(std::string Path) override;
+  std::optional<std::string> writeFeatureModel();
+
+private:
+  using constXmlCharPtr = const xmlChar *;
+
+  const FeatureModel &Fm;
```

Comment:
regarding the style guide, we should move member vars to the bottom of the private section